File: backend/app/schemas/reader.py
from pydantic import BaseModel
from typing import Optional


class SurahInfo(BaseModel):
    number: int
    name_arabic: str
    name_english: str
    name_transliteration: Optional[str] = None
    name_russian: Optional[str] = None          # перевод названия ("Весть")
    name_transliteration_ru: Optional[str] = None  # транслит ("Ан-Наба")
    revelation_type: Optional[str] = None
    total_ayahs: int

    model_config = {"from_attributes": True}


class WordInAyah(BaseModel):
    word_id: Optional[int] = None
    position: int
    arabic: str
    translation_ru: Optional[str] = None  # первое значение слова для подписи
    is_in_study: bool = False


class AyahWithWords(BaseModel):
    id: int
    surah_number: int
    ayah_number: int
    arabic_text: str
    russian_translation: Optional[str] = None
    words: list[WordInAyah]

    model_config = {"from_attributes": True}


class EnqueueRequest(BaseModel):
    word_ids: list[int]


class EnqueueResponse(BaseModel):
    added: int
    already_studying: int

File: frontend/src/api/reader.ts
import { apiClient } from './client'

export interface SurahInfo {
  number: number
  name_arabic: string
  name_english: string
  name_transliteration: string | null
  name_russian: string | null
  name_transliteration_ru: string | null
  revelation_type: string | null
  total_ayahs: number
}

export interface WordInAyah {
  word_id: number | null
  position: number
  arabic: string
  translation_ru: string | null
  is_in_study: boolean
}

export interface AyahWithWords {
  id: number
  surah_number: number
  ayah_number: number
  arabic_text: string
  russian_translation: string | null
  words: WordInAyah[]
}

export interface EnqueueResponse {
  added: number
  already_studying: number
}

export const readerApi = {
  getSurahs: () =>
    apiClient.get<SurahInfo[]>('/reader/surahs'),

  getAyahs: (surahNumber: number) =>
    apiClient.get<AyahWithWords[]>(`/reader/ayahs/${surahNumber}`),

  enqueue: (wordIds: number[]) =>
    apiClient.post<EnqueueResponse>('/reader/enqueue', { word_ids: wordIds }),

  dequeue: (wordId: number) =>
    apiClient.delete(`/reader/words/${wordId}`),
}
